### PR TITLE
add grantReward ToEthTx()

### LINKED ISSUE #4046: web3 RPC to support BlockReward type transaction
### Description of new feature request
for integration with https://chainspect.app/dashboard/tps. They want to eliminate any non-user transaction from a block to accurately calculate the real TPS of a blockchain. So they want to identify any protocol-initiated tx

We can convert the BlockReward tx, with `ContractAddress` field in the receipt equals to the protocol address `0xA576C141e5659137ddDa4223d209d4744b2106BE`. This would allow the receiving client to uniquely identify the BlockReward tx

### Is this a hard-fork (and height to be activated)
- [] Yes
- Name of activation height:

### Implementation plan
- [] Phase 1 description (PR number)
- [] Phase 2 description (PR number)
- [] Phase 3 description (PR number)

### Affected modules
- [x] action
- [] actpool
- [x] api
- [] blockchain
- [] blockindex
- [] blocksync
- [] chainservice
- [] config
- [] consensus
- [] db
- [] dispatcher
- [] ioctl
- [] p2p
- [] server
- [] state
- [] tools

### PR DESCRIPTION
**Pull request by millken**
# Description
when the action type is `grantRward`, eth client will get an error `the type of action is not supported`
```
$ curl -s -H "Content-Type: application/json" -X POST --data '{"jsonrpc":"2.0","method":"eth_getTransactionByHash","params":["0x69b9836c7717d08e73d8b22a3a1a69b59261ba4c64c86607cf331eb22aaffc1c"],"id":1}' https://babel-api.mainnet.iotex.io
{"jsonrpc":"2.0","id":1,"error":{"code":-32603,"message":"actHash: 69b9836c7717d08e73d8b22a3a1a69b59261ba4c64c86607cf331eb22aaffc1c: the type of action is not supported","data":null}}%
```

Fixes #4046 

## Type of change
Please delete options that are not relevant.
- [x] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: action/grantreward.go [hunk 1]
```diff
@@ -78,6 +79,17 @@ func (*GrantReward) Cost() (*big.Int, error) {
 	return big.NewInt(0), nil
 }
 
+// ToEthTx converts a grant reward action to an ethereum transaction
+func (g *GrantReward) ToEthTx(_ uint32) (*types.Transaction, error) {
```
[the remark this answers is not in the document]
Comment (millken) [reply action/grantreward.go:83]: grantReward is native type, not contract, For external users, just know that to = 'rewardingaddr' is a special action

[the remark this answers is not in the document]
Comment (CoderZhi) [reply action/grantreward.go:83]: may be include `height` and `reward type` in payload

**Comment by Liuhaai** [comment]
add `EncodeABIBinary()` and refactor the tests

**Review by dustinxie — APPROVED** [review]

**Comment by millken** [comment]
> add `EncodeABIBinary()` and refactor the tests

updated

**Pull request merged**